### PR TITLE
Adds a food trash element

### PR DESCRIPTION
**Pull request by ghost**
:cl: qustinnus
code: food trash element, for dumping trash when u eat food
/:cl:

unused now, needed for food refactor. spawns trash in your hands (or on the floor) when the food this is attached to is eaten.

### CODE REVIEW

File: code/datums/elements/food/food_trash.dm [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	var/trash
+
+/datum/element/food_trash/Attach(datum/target, atom/trash)
+	. = ..()
+	if(!isatom(target))
+		return ELEMENT_INCOMPATIBLE
+	src.trash = trash
+	RegisterSignal(target, COMSIG_FOOD_CONSUMED, .proc/generate_trash)
+
+/datum/element/dunkable/Detach(datum/target)
```
Comment (Donkie) [inline code/datums/elements/food/food_trash.dm:15]: Whoopsie

**Review by Donkie — CHANGES_REQUESTED** [review]

File: code/datums/elements/food/food_trash.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	. = ..()
+	UnregisterSignal(target, COMSIG_FOOD_CONSUMED)
+
+/datum/element/food_trash/proc/generate_trash(datum/source, mob/living/eater, mob/living/feeder)
+	SIGNAL_HANDLER
```
Comment (LemonInTheDark) [inline code/datums/elements/food/food_trash.dm:20]: you're sleeping in this proc, you run the risk of raising the ire of the dreaded moth maintainer
image

Comment (LemonInTheDark) [reply code/datums/elements/food/food_trash.dm:20]: Something is calling/can call under the right circumstances do_mob 
image

Comment (Qustinnus) [reply code/datums/elements/food/food_trash.dm:20]: wtf ill have to look at the chain for that later

Comment (Qustinnus) [reply code/datums/elements/food/food_trash.dm:20]: what a dumb ass call chain

Comment (LemonInTheDark) [reply code/datums/elements/food/food_trash.dm:20]: Mood

**Pull request merged**